WordPress/gutenberg
# [Mobile] - Search block - Colors support

- Gutenberg Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/4127

## Description
Recently color support [was added](https://github.com/WordPress/gutenberg/pull/32416) for the Search block, this PR adds support on mobile.

## How has this been tested?

### Test case 1 - Default settings

- Open the app
- Add the Search block
- **Expect** to see the default colors:
  - Default editor colors without a block-based theme
  - Colors of the theme for block-based themes

### Test case 2 - Theme colors

- Open the app (with a block-based theme activated)
- Add the Search block
- Open the block settings
- Change the text and background colors (using the theme's palette)
- **Expect** to see the search button with the selected colors

### Test case 3 - Custom colors

- Open the app (with a block-based theme activated)
- Add the Search block
- Open the block settings
- Change the text and background colors (using **custom** colors with the color picker)
- **Expect** to see the search button with the selected colors

### Test case 4 - Icon color

- Open the app (with a block-based theme activated)
- Add the Search block
- Open the block settings
- Enable the option: `Use icon button`
- Close the settings
- **Expect** to see the color of the icon to be the same as the text color
- Open the block settings
- Change the text color using a color from the theme's palette or using the color picker
- **Expect** to see the icon with the selected color

## Screenshots 
iOS |Android
-|-
<kbd><img src="https://user-images.githubusercontent.com/4885740/136791431-41ef7975-370a-4fb9-802e-532346b9d286.png" width="200" /></kbd> | <kbd><img src="https://user-images.githubusercontent.com/4885740/136791456-d9694bc6-87ac-4d54-abfd-d3dffed999b8.png" width="200" /></kbd>

## Types of changes
New feature

## Checklist:
